lib/internal/cluster/round_robin_handle.js

```diff
@@ -84,32 +98,46 @@ RoundRobinHandle.prototype.remove = function(worker) {
 
 RoundRobinHandle.prototype.distribute = function(err, handle) {
   this.handles.push(handle);
-  const worker = this.free.shift();
-
-  if (worker)
-    this.handoff(worker);
+  this.handoff();
 };
 
-RoundRobinHandle.prototype.handoff = function(worker) {
-  if (worker.id in this.all === false) {
-    return;  // Worker is closing (or has closed) the server.
+RoundRobinHandle.prototype.handoff = function() {
+  const handle = this.handles[0];
+  var socket;
+
+  // There are currently no requests to schedule.
+  if (handle === undefined)
+    return;
+
+  if (this.scheduler.exposeSocket === true) {
+    socket = new net.Socket({
+      handle,
+      readable: false,
+      writable: false,
+      pauseOnCreate: true
+    });
   }
 
-  const handle = this.handles.shift();
+  const worker = this.scheduler(this.workers, socket);
```

Comment:
The docs currently say:

> The scheduling function should return the worker that will handle the connection. However, the worker should not be removed from `workers`.

I think slicing would reduce the chance of shooting yourself in the foot, but it would make it more work to do something like round robin because we modify the array order in the scheduler.